go/client/cmd_deprovision.go

```diff
@@ -0,0 +1,124 @@
+	rpc "github.com/keybase/go-framed-msgpack-rpc"
+	"golang.org/x/net/context"
+)
+
+type CmdDeprovision struct{}
+
+func (c *CmdDeprovision) Run() (err error) {
+	protocols := []rpc.Protocol{
+		NewLogUIProtocol(),
+		NewSecretUIProtocol(G),
+	}
+	if err = RegisterProtocols(protocols); err != nil {
+		return err
+	}
+
+	username, err := c.getUsernameToDeprovision()
+	if err != nil {
+		return err
+	}
+
+	if err = GlobUI.PromptForConfirmation(getBigScaryWarning(username)); err != nil {
+		return
+	}
+
+	loginCli, err := GetLoginClient(G)
+	if err != nil {
+		return err
+	}
+	// XXX: This RPC deletes secret keys!
+	return loginCli.Deprovision(context.TODO(), keybase1.DeprovisionArg{
+		SessionID: 0,
+		Username:  username,
+	})
+}
+
+func NewCmdDeprovision(cl *libcmdline.CommandLine) cli.Command {
+	return cli.Command{
+		Name:  "deprovision",
+		Usage: "Revoke the current device, log out, and delete local state.",
+		Action: func(c *cli.Context) {
+			cl.ChooseCommand(&CmdDeprovision{}, "deprovision", c)
+		},
+	}
+}
+
+func (c *CmdDeprovision) GetUsage() libkb.Usage {
+	return libkb.Usage{
+		Config: true,
+		API:    true,
+	}
+}
+
+func (c *CmdDeprovision) ParseArgv(*cli.Context) error { return nil }
+
+func (c *CmdDeprovision) getUsernameToDeprovision() (string, error) {
+	configCli, err := GetConfigClient(G)
+	if err != nil {
+		return "", err
+	}
+	currentStatus, err := configCli.GetCurrentStatus(context.TODO(), 0)
+	if err != nil {
+		return "", err
+	}
+
+	// If there is a user logged in, just return that user.
+	if currentStatus.LoggedIn {
+		return currentStatus.User.Username, nil
+	}
+
+	// Otherwise, find all the users in the config file.
+	current, others, err := G.Env.GetConfig().GetAllUsernames()
+	if err != nil {
+		return "", err
+	}
+	allUsers := []libkb.NormalizedUsername{}
+	if current.String() != "" {
+		allUsers = append(allUsers, current)
+	}
+	allUsers = append(allUsers, others...)
+
+	// If there's no one in the config file, there's nothing to do.
+	if len(allUsers) == 0 {
+		return "", fmt.Errorf("Can't find a user to deprovision.")
+	}
+
+	// If there's only one user in the config file, just pick that one.
+	if len(allUsers) == 1 {
+		return allUsers[0].String(), nil
+	}
+
+	fmt.Printf("No one currently logged in. Which user would you like to deprovision?\n")
+	for i, user := range allUsers {
+		fmt.Printf("%d) %s\n", i+1, user.String())
+	}
+	choice, err := PromptSelectionOrCancel(PromptDescriptorDeprovisionWhichUser,
+		G.UI.GetTerminalUI(), "Choose a user", 1, len(allUsers))
+	if err != nil {
+		return "", err
+	}
+	return allUsers[choice-1].String(), nil
+}
+
+func getBigScaryWarning(username string) string {
+	// TODO: Print a list of the other devices on the user's account.
+	return fmt.Sprintf(`
+%s, BE CAREFUL!  \('o')/
+
+You are about to delete this device from your account, including its secret
+keys. If you don't have any other devices, you'll lose access to your account
+and all your data!
```

Comment:
This big scary warning should say whether they have PGP keys in their secret keyring, and whether they should be nuked too.  If not, then please export via `keybase pgp export`